### PR TITLE
Makefile locale handling

### PR DESCRIPTION
**Pull request by coldtobi**
Proposal how to handle locales 

as described in #249: quoted for convenience:

> My plan of attack would be:
> - add a rule to generate the pot file*
> - add rules to generate the mo files at build time
> - subsequently removing the mo files from the repository
> 
> (* as you know the pot file needs updating and cannot be regenerated from scratch without losing all translated texts… and git does not preserve timestamp. So it would be safer to run the pot generation always or use some stamp file)
> with some kind of stamp file.)

(I've intentionally made small commits to explain what my thoughts were in the commit messages. Of course, I can squash them, if you want)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+
+distclean: clean
+	rm -rf hidetopbar.zip locale/locale/hidetop.pot-stamp
+
+%.mo: %.po locale/hidetopbar.pot
+	msgfmt -c -o $@ $<
+
+locale/hidetopbar.pot locale/hidetop.pot-stamp : $(JS_FILES)
+	xgettext --output=./locale/hidetopbar.pot --language=JavaScript $^
+	touch locale/hidetop.pot
```
Comment (tuxor1337) [inline Makefile:29]:
```suggestion
	touch locale/hidetopbar.pot-stamp
```

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
-	zip hidetopbar.zip -r COPYING.txt $(JS_FILES) metadata.json locale/*/*/*.mo schemas
+hidetopbar.zip: schemas/gschemas.compiled $(LOCALES_MO)
+	zip hidetopbar.zip -r COPYING.txt $(JS_FILES) metadata.json $(LOCALES_MO) schemas
 
 clean:
-	rm -rf hidetopbar.zip schemas/gschemas.compiled
+	rm -rf hidetopbar.zip schemas/gschemas.compiled ${LOCALES_MO}
+
+distclean: clean
+	rm -rf hidetopbar.zip locale/locale/hidetop.pot-stamp
```
Comment (tuxor1337) [inline Makefile:22]:
```suggestion
	rm -rf hidetopbar.zip locale/locale/hidetopbar.pot-stamp
```

File: Makefile [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
+
+%.mo: %.po locale/hidetopbar.pot
+	msgfmt -c -o $@ $<
+
+locale/hidetopbar.pot locale/hidetop.pot-stamp : $(JS_FILES)
```
Comment (tuxor1337) [inline Makefile:27]:
```suggestion
locale/hidetopbar.pot locale/hidetopbar.pot-stamp : $(JS_FILES)
```

**Review by tuxor1337 — CHANGES_REQUESTED** [review]
Great proposal! Looks to me like there is some inconsistency in the naming of the stamp file. Apart from that, it looks good. And yes, please squash. It's clear enough in one commit, I think.

**Comment by coldtobi** [comment]
Fixed. Addtionally fixed 2 minor things:
- stamp was missing as prerequesite for the mo rule
- distclean did also rm hidetopbar.zip, which was already cleaned in clean.

**Pull request merged**